common/lessonplan/generate.ts

```
@@ -0,0 +1,298 @@
+                        loader = new DocxLoader(blob);
+                        docs = await loader.load();
+                    } else if (file.mimetype === 'application/vnd.openxmlformats-officedocument.presentationml.presentation') {
+                        loader = new PPTXLoader(blob);
+                        docs = await loader.load();
+                    } else {
+                        throw new Error(`Unsupported file type: ${file.mimetype}`);
+                    }
+
+                    // Add filename and page numbers (only for PDFs) to the content
+                    const content =
+                        `Filename: ${file.originalname}\n\n` +
+                        docs
+                            .map((doc, index) => (file.mimetype === 'application/pdf' ? `Page ${index + 1}:\n${doc.pageContent}` : doc.pageContent))
+                            .join('\n\n');
+
+                    // Check if the content is empty (excluding filename and page numbers)
+                    const contentWithoutMetadata = content
+                        .replace(`Filename: ${file.originalname}\n\n`, '')
+                        .replace(/Page \d+:\n/g, '')
+                        .trim();
+                    if (contentWithoutMetadata.length === 0) {
+                        emptyFiles.push(file.originalname);
+                        return null;
+                    }
+
+                    return {
+                        name: file.originalname,
+                        content: content,
+                        type: file.mimetype,
+                    };
+                } catch (error) {
+                    logger.error(`Error fetching or parsing file with UUID ${uuid}: ${error.message}`);
+                    throw error;
+                }
+            })
+        );
+
+        const validFiles = allFiles.filter((file): file is NonNullable<typeof file> => file !== null);
+
+        if (emptyFiles.length > 0) {
```

Why not just `throw` instead of `return null` above? Same for the empty files ...

---

`emptyFiles` is now never written

---

Could still remove the `emptyFiles` variable